sub_peers.js

```diff
@@ -125,7 +197,16 @@ TChannelSubPeers.prototype.chooseLinearPeer = function chooseLinearPeer(req) {
     for (var i = 0; i < hosts.length; i++) {
         var hostPort = hosts[i];
         var peer = self._map[hostPort];
-        if (!req || !req.triedRemoteAddrs || !req.triedRemoteAddrs[hostPort]) {
+
+        var shouldSkip = req && req.triedRemoteAddrs && req.triedRemoteAddrs[hostPort];
+        if (this.hasMinConnections) {
+            var notEnoughPeers = this.currentConnectedPeers < this.minConnections;
```

Comment:
We could but that increases complexity. 

If `minConnections=10` it will warm up pretty quick. The window of time where you wait for 1 init handshake or 10 init handshakes is similar.

I think its a micro op that we can do later.